src/components/course/enrollment/utils.js

```diff
@@ -15,31 +15,54 @@ const {
   HIDE_BUTTON,
 } = enrollButtonTypes;
 
+/**
+ * Determines whether a user can request a subsidy for a course, by checking whether
+ * the subsidy request feautre is enabled, there are more than 1 subsidy request
+ * catalogs applicable to the course, and whether the learner already has a subsidy
+ * applicable to the course.
+ *
+ * @param {object} args
+ * @param {object} args.subsidyRequestConfiguration Contains a property whether subsidy requests are enabled
+ * @param {Set} args.subsidyRequestCatalogsApplicableToCourse Set representing the subsidy catalogs
+ *  applicable to the course
+ * @param {boolean} args.userSubsidyApplicableToCourse Subsidy applicable to the course
+ *
+ * @returns True if the user can request a subsidy for the course, false otherwise.
+ */
+export function canUserRequestSubsidyForCourse({
+  subsidyRequestConfiguration,
+  subsidyRequestCatalogsApplicableToCourse,
+  userSubsidyApplicableToCourse,
+}) {
+  // Hide enroll button if browse and request is turned on and the user has no applicable subsidy
+  if (!subsidyRequestConfiguration || !subsidyRequestCatalogsApplicableToCourse) {
+    return false;
+  }
+  return (
+    subsidyRequestConfiguration.subsidyRequestsEnabled
+    && subsidyRequestCatalogsApplicableToCourse.size > 0
+    && !userSubsidyApplicableToCourse
```

Comment:
@marlonkeating The former, "indicates that the user already has a subsidy and doesn't need to request one". This logic was just moved from another place to here, but is otherwise unchanged.